e2e/src/l2.spec.ts

```diff
@@ -1,339 +1,321 @@
-import { Wallet, ethers } from "ethers";
-import { describe, expect, it } from "@jest/globals";
-import { TransactionRequest } from "@ethersproject/providers";
-import { getAndIncreaseFeeData } from "./utils/helpers";
-import { RollupGetZkEVMBlockNumberClient, getEvents, wait } from "./utils/utils";
-
-const layer2TestSuite = (title: string) => {
-  describe(title, () => {
-    describe("Transaction data size", () => {
-      it("Should revert if transaction data size is above the limit", async () => {
-        const account = new Wallet(L2_DEPLOYER_ACCOUNT_PRIVATE_KEY, l2Provider);
-        await expect(
-          dummyContract.connect(account).setPayload(ethers.utils.randomBytes(TRANSACTION_CALLDATA_LIMIT)),
-        ).rejects.toThrow("err: tx data is too large (in bytes)");
-      });
-
-      it("Should succeed if transaction data size is below the limit", async () => {
-        const account = new Wallet(L2_ACCOUNT_0_PRIVATE_KEY, l2Provider);
+import { JsonRpcProvider, TransactionRequest, ethers } from "ethers";
+import { beforeAll, describe, expect, it } from "@jest/globals";
+import { config } from "./config/tests-config";
+import { getAndIncreaseFeeData } from "./common/helpers";
+import { RollupGetZkEVMBlockNumberClient, etherToWei, getEvents, wait } from "./common/utils";
+import { TRANSACTION_CALLDATA_LIMIT } from "./common/constants";
+
+describe("Layer 2 test suite", () => {
+  let l2Provider: JsonRpcProvider;
+
+  beforeAll(() => {
+    l2Provider = config.getL2Provider();
+  });
 
-        const [nonce, feeData] = await Promise.all([
-          l2Provider.getTransactionCount(account.address),
-          l2Provider.getFeeData(),
-        ]);
+  it.concurrent("Should revert if transaction data size is above the limit", async () => {
+    const account = await config.getL2AccountManager().generateAccount();
+    const dummyContract = config.getL2DummyContract(account);
 
-        const [maxPriorityFeePerGas, maxFeePerGas] = getAndIncreaseFeeData(feeData);
-        const tx = await dummyContract.connect(account).setPayload(ethers.utils.randomBytes(1000), {
-          nonce,
-          maxPriorityFeePerGas,
-          maxFeePerGas,
-        });
+    await expect(
+      dummyContract.connect(account).setPayload(ethers.randomBytes(TRANSACTION_CALLDATA_LIMIT)),
+    ).rejects.toThrow("missing revert data");
+  });
 
-        const receipt = await tx.wait();
-        expect(receipt.status).toEqual(1);
-      });
+  it.concurrent("Should succeed if transaction data size is below the limit", async () => {
```

Comment:
I think this test can be combined with the next test below: `Should succeed in conflating transactions with large calldata with low gas into multiple L1 blocks`, we're basically the same with different transactions, and I dun see the benefit of separating them, anyway up to you

Reply:
Or we can remove them as well, because they don't actually test conflation much nor we can get a deterministic conflation behaviour in E2E test